planner/core/casetest/integration_test.go

```diff
@@ -65,6 +65,22 @@ func TestPushLimitDownIndexLookUpReader(t *testing.T) {
 	}
 }
 
+func TestHintInWrongPos(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+
+	tk.MustExec("set tidb_cost_model_version=2")
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b int)")
+
+	tk.MustExec("select /*+ stream_agg() */ count(*) from t where a > 1;")
+	tk.MustQuery("show warnings").Check(testkit.Rows())
+
+	tk.MustExec("select count(*) /*+ stream_agg() */ from t where a > 1;")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1064 You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use [parser:8066]Optimizer hint can only be followed by certain keywords like SELECT, INSERT, etc."))
```

Comment:
Can this warning take effect in `insert into  ... select /*+ hint()*/ ...` and subquery